### PR TITLE
Display removed panels for commoners

### LINKED ISSUE #4696: Allow toggling display of hidden panels for normal users
Conversation continued from #4695

@Jakob37 writes:

> A separate but related point which I am not sure whether an issue or not, is that while I can see the removed panels using "Hide removed panels", our lead geneticsts could not. When she clicked that button for the same institute for which I could see removed panels, nothing happened for her. She tested also in my browser with her account, so seems to be a user setting.
> 
> Is it only admin that could see the removed panels? It would be useful for her to be able to see them.

@dnil writes:

> Not being able to see the removed panels is at least partly apparently by design; the same `shall_display_panel()` is used, so if a panel is not visible, a user needs to have write access, which they only have to panels they are `maintainer` for, unless they are admins. I see no harm in showing these on request: on the panelS page, the option indicates they can already select if they want that extra clutter! They should still not be able to restore it without "write" permissions for it naturally. But yeah, let's make a new issue to keep the conversations easier to follow! 😅

### PR DESCRIPTION
**Pull request by northwestwitch**
Closes #4696 

I'm wondering if the non-admin, non-maintainer user should be able to modify (add/remove genes thus creating a new panel) a removed panel? 

## Testing

What happens when a user that is not admin toggles the hide remove panels on panels page:

image

## Main branch: - note that removed panels are already visible if the user is a maintainer

image

### View all panels, also those that are hidden
image

<hr>

### Added a Maintainer badge:

image

<hr>

### Disabled "remove" and "restore" panel buttons for those panels the users lacks privileges for:

image

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by Jakob37
- [x] tests executed by CR, Jakob37

### CODE REVIEW

**Comment by northwestwitch** [comment]
Hi @Jakob37, I have a PR to fix your issue here. You could check it out, when you have time!

**Comment by Jakob37** [comment]
> Hi @Jakob37, I have a PR to fix your issue here. You could check it out, when you have time!

Great, thanks!

Will go for vacation for another two weeks after this, so will try taking a look today.

**Comment by northwestwitch** [comment]
> Great, thanks!
> 
> Will go for vacation for another two weeks after this, so will try taking a look today.

Nice! No worries, it's fine also after the holidays :)

**Comment by Jakob37** [comment]
>Nice! No worries, it's fine also after the holidays :)

My computer broke down today, so I'll have to pick up on that offer. Will be back after holiday 🌴

**Comment by Jakob37** [comment]
I tested out this PR. Looks great!

I did the following:

- Added a regular user
- Added three panels
- Removed two
- Added the user as maintainer for two panels, one removed and one non-removed
- Logged into the user and tried removing / restoring
- Clicking the "show hidden"

What happened:

- When logging in, I could first see two panels, the ones not removed.
- When clicking "show hidden", all four panels showed up, two of them grayed out (see screenshot)
- Only non grayed out buttons worked, as expected

reg_user_maintenance_of_panels

Nice job!

File: scout/server/blueprints/panels/templates/panels/panels.html [hunk 1]
```diff
@@ -269,6 +272,11 @@ <h5 class="modal-title">Remove gene panel "{{panel.display_name}}"?</h5>
 <script src="https://cdn.datatables.net/1.12.0/js/jquery.dataTables.min.js" integrity="sha512-fu0WiDG5xqtX2iWk7cp17Q9so54SC+5lk/z/glzwlKFdEOwGG6piUseP2Sik9hlvlmyOJ0lKXRSuv1ltdVk9Jg==" referrerpolicy="no-referrer" crossorigin="anonymous"></script>
 <script type="text/javascript">
 
+var tooltipTriggerList = [].slice.call(document.querySelectorAll('[data-bs-toggle="tooltip"]'))
+var tooltipList = tooltipTriggerList.map(function (tooltipTriggerEl) {
```
Comment (Jakob37) [inline scout/server/blueprints/panels/templates/panels/panels.html:276]: Do you need to retain these variables? Or would it work to:

```
document.querySelectorAll('[data-bs-toggle="tooltip"]').forEach(function (tooltipTriggerEl) {
  new bootstrap.Tooltip(tooltipTriggerEl);
});
```

If the aim is to just initiate the tooltips. Just curious!

**Review by Jakob37 — APPROVED** [review]
Great 👍

File: scout/server/blueprints/panels/templates/panels/panels.html (hunk 1 above)
Comment (northwestwitch) [reply scout/server/blueprints/panels/templates/panels/panels.html:276]: I think I might have simply copied it from another page. I'll test what you suggest

Comment (northwestwitch) [reply scout/server/blueprints/panels/templates/panels/panels.html]: Your were right, it's not needed. I removed it now. Thanks!

**Pull request merged**